overlay.d/05core/usr/lib/dracut/modules.d/20live/live-generator

```
@@ -15,7 +15,7 @@ add_requires() {
     ln -sf "../${name}" "${requires_dir}/${name}"
 }
 
-isoroot=$(getarg coreos.liveiso=)
+isoroot=$(getarg coreos.liveiso= 2>/dev/null || true)
```

Is it worth moving this into the `else` clause instead of discarding the error? The generator should probably fail if `coreos.liveiso` isn't set and you are trying to boot an iso, right?

---

Good point.  I went with a bigger cleanup, moving up the check and exiting early, allowing us to de-indent everything else :arrow_down:

---

I mean, even with the bigger cleanup, this line still has to be in the `else` clause, right? It should only be specified when booting an ISO, not when booting from PXE

---

There isn't a need for an `else` clause - we exit the whole generator early on if we're not in a live image.  Right?

---

I'm sorry, now that I re-read this I understand now - you were talking about the *inner* condition and it indeed makes total sense to move the `getarg` there.  Thanks so much for your review!